### PR TITLE
Access network info through API, instead of directly fetching attributes

### PR DESCRIPTION
**Pull request by vuntz**
This will help have the network info in only one location.

Should go with https://github.com/crowbar/crowbar-core/pull/510

### CODE REVIEW

File: chef/cookbooks/neutron/recipes/server.rb [hunk 1]
```diff
@@ -91,7 +91,8 @@
   only_if { node[:platform_family] == "debian" }
 end
 
-vlan_start = node[:network][:networks][:nova_fixed][:vlan]
+nova_fixed_net = Barclamp::Inventory.get_network_by_type(node, "nova_fixed")
```
Comment (rhafer) [inline chef/cookbooks/neutron/recipes/server.rb]: Don't you also need to access this directly  via the roles here? "nova_fixed" will not be enabled on the node when only the gre and/or vxlan drivers are enabled.

Comment (vuntz) [inline chef/cookbooks/neutron/recipes/server.rb]: You're right; I forgot to fix that back (I had the issue in `post_install_conf`).

Comment (vuntz) [inline chef/cookbooks/neutron/recipes/server.rb]: Fixed.

**Comment by jsuchome** [comment]
Could you remove it from product, so the gating can pass?

**Comment by vuntz** [comment]
> Could you remove it from product, so the gating can pass?

(for the record, since we discussed on irc) It's not in the package, I guess we're hitting a bug with jenkins that somehow reuses the package from the gating job :/

**Comment by toabctl** [comment]
+1

**Comment by rhafer** [comment]
+1, but AFAICS this needs to wait for https://github.com/crowbar/crowbar-core/pull/510 to be merged

**Pull request merged**